contracts/TranscodeJobs.sol

```diff
@@ -0,0 +1,141 @@
+            return JobStatus.Inactive;
+        } else {
+            // A job is active if the current block is less than the job's termination block
+            return JobStatus.Active;
+        }
+    }
+
+    function newJob(Jobs storage self, uint256 _streamId, bytes32 _transcodingOptions, uint256 _maxPricePerSegment, address _electedTranscoder) returns (bool) {
+        self.jobs[self.numJobs].jobId = self.numJobs;
+        self.jobs[self.numJobs].streamId = _streamId;
+        self.jobs[self.numJobs].transcodingOptions = _transcodingOptions;
+        self.jobs[self.numJobs].maxPricePerSegment = _maxPricePerSegment;
+        self.jobs[self.numJobs].broadcasterAddress = msg.sender;
+        self.jobs[self.numJobs].transcoderAddress = _electedTranscoder;
+        self.numJobs++;
+
+        return true;
+    }
+
+    function getJob(Jobs storage self, uint256 _jobId) constant returns (uint256, uint256, bytes32, uint256, address, address, uint256, uint256, uint256) {
+        Job job = self.jobs[_jobId];
+
+        return (job.jobId,
+                job.streamId,
+                job.transcodingOptions,
+                job.maxPricePerSegment,
+                job.broadcasterAddress,
+                job.transcoderAddress,
+                job.endBlock,
+                job.lastClaimedWorkBlock,
+                job.endVerificationBlock
+                );
+    }
+
+    function getJobWorkDetails(Jobs storage self, uint256 _jobId) constant returns (uint256, uint256, bytes32) {
+        Job job = self.jobs[_jobId];
+
+        bytes32 lastTranscodeClaimRoot = 0x0;
+
+        if (job.transcodeClaimsRoots.length > 0) {
+            lastTranscodeClaimRoot = job.transcodeClaimsRoots[job.transcodeClaimsRoots.length - 1];
+        }
+
+        return (job.lastClaimedSegmentRange[0], job.lastClaimedSegmentRange[1], lastTranscodeClaimRoot);
+    }
+
+    function endJob(Jobs storage self, uint256 _jobId, uint256 _jobEndingPeriod) returns (bool) {
+        // Check if job already has an end block
+        if (self.jobs[_jobId].endBlock > 0) throw;
+        // Check if called by either broadcaster or transcoder
+        if (msg.sender != self.jobs[_jobId].broadcasterAddress && msg.sender != self.jobs[_jobId].transcoderAddress) throw;
+
+        // Set set end block for job
+        self.jobs[_jobId].endBlock = block.number + _jobEndingPeriod;
+
+        return true;
+    }
+
+    function claimWork(Jobs storage self, uint256 _jobId, uint256 _startSegmentSequenceNumber, uint256 _endSegmentSequenceNumber, bytes32 _transcodeClaimsRoot, uint256 _verificationPeriod) returns (bool) {
+        // Check if job is active
+        if (jobStatus(self, _jobId) != JobStatus.Active) throw;
+        // Check if sender is assigned transcoder
+        if (self.jobs[_jobId].transcoderAddress != msg.sender) throw;
+        // Check if previous verification period over
+        if (block.number < self.jobs[_jobId].endVerificationBlock) throw;
```

Comment:
For now we force transcoders to call verify a call to `claimWork()` before they can make a subsequent call to `claimWork()`